src/main/scala/org/neo4j/spark/util/Validations.scala

```diff
@@ -15,6 +15,7 @@ object Validations {
     val cache = new DriverCache(neo4jOptions.connection, jobId)
     val schemaService = new SchemaService(neo4jOptions, cache)
     try {
+      validateConnection(cache.getOrCreate().session(neo4jOptions.session.toNeo4jSession))
```

Comment:
watch out on semantics; this method is saying "get or create" - is session reuse possible here?  Because validateConnection closes the session.   Not sure if there's a code path to be concerned about there, but you have to throw away this session after validating the connection and you can't reuse it.

Reply:
`getOrCreate` is referred to the DriverCache, so what we're getting here is a driver instance and it's reusable. We can clear it up by chancing the `cache` variabile to `driverCache`